text/0000-local-snapshot-file-format/0000-local-snapshot-file-format.md

```diff
@@ -0,0 +1,149 @@
+
+A full ledger snapshot file contains the UTXOs (`outputs` section) of a node's latest solid milestone (`ledger_milestone_index`). The `diffs` contain the diffs to rollback the `outputs` state in order to regain the ledger state of the snapshot milestone at (`seps_milestone_index`).
+
+![](https://i.imgur.com/e6WuufK.png)
+
+While the node producing such full ledger state snapshot could theoretically pre-compute the actual snapshot milestone state, this is deferred to the consumer of the data in order to speed up local snapshot creation.
+
+A full ledger state local snapshot is denoted by the type byte `0`:
+
+```
+version<byte>
+type<byte> = 0
+timestamp<uint64>
+seps_milestone_index<uint64>
+seps_milestone_hash<array[32]>
+ledger_milestone_index<uint64>
+ledger_milestone_hash<array[32]>
+seps_count<uint64>
+outputs_count<uint64>
+diffs_count<uint64>
+seps<array[seps_count]>:
+	sep<array[32]>
+outputs<array[outputs_count]>:
+	transaction_hash<array[32]>
+	output_index<uint16>
+	address:
+		address_type<byte>
+		wots_address<array[49]>
+		||
+		address_type<byte>
+		ed25519_address<array[32]>
+	value<uint64>
+diffs<array[diffs_count]>:
+	diff_milestone_index<array[32]>
+	created_outputs_count<uint64>
+	created_outputs<array>:
+		transaction_hash<array[32]>
+		output_index<uint16>
+		address:
+			address_type<byte> = 0
+			wots_address<array[49]>
+			||
+			address_type<byte> = 1
+			ed25519_address<array[32]>
+		value<uint64>
+	consumed_outputs_count<uint64>
+	consumed_outputs<array>:
+		transaction_hash<array[32]>
+		output_index<uint16>
+		address:
+			address_type<byte>
+			wots_address<array[49]>
+			||
+			address_type<byte>
+			ed25519_address<array[32]>
+		value<uint64>		
+```
+
+#### Delta Ledger State
+
+A delta ledger state local snapshot only contains the `diffs` of milestones starting from a given `ledger_milestone_index`. A node consuming such data must know the state of the ledger at `ledger_milestone_index`.
+
+![](https://i.imgur.com/bt5BUpe.png)
+
+A delta ledger state local snapshot is denoted by the type byte `1`:
+
+```
+version<byte>
+type<byte> = 1
+timestamp<uint64>
+seps_milestone_index<uint64>
+seps_milestone_hash<array[32]>
+ledger_milestone_index<uint64>
+ledger_milestone_hash<array[32]>
+seps_count<uint64>
+diffs_count<uint64>
+seps<array[seps_count]>:
+	sep<array[32]>
+diffs<array[diffs_count]>:
+	diff_milestone_index<array[32]>
+	created_outputs_count<uint64>
+	created_outputs<array>:
+		transaction_hash<array[32]>
+		output_index<uint16>
+		address:
+			address_type<byte> = 0
+			wots_address<array[49]>
+			||
+			address_type<byte> = 1
+			ed25519_address<array[32]>
+		value<uint64>
+	consumed_outputs_count<uint64>
+	consumed_outputs<array>:
+		transaction_hash<array[32]>
+		output_index<uint16>
+		address:
+			address_type<byte> = 0
+			wots_address<array[49]>
+			||
+			address_type<byte> = 1
+			ed25519_address<array[32]>
+		value<uint64>
+```
+
+# Drawbacks
+
+Nodes need to support this new format.
```

Comment:
I guess you had nothing to fill in :-) ?

Maybe:
1. We have 2 different snapshot files which may be confusing
2. The size of the file will grow as the number of outputs increase